lisa/nic.py

```diff
@@ -0,0 +1,222 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import ipaddress
+import itertools
+import re
+from typing import Dict, List
+
+from assertpy import assert_that
+
+from lisa import Node
+
+
+class NicInfo:
+
+    # Class for info about an single upper/lower nic pair.
+    # devices using SRIOV on azure typically have an upper synthetic device
+    # paired with a lower SRIOV Virtual Function (VF) device that
+    # enables the passthrough to the physical NIC.
+
+    def __init__(
+        self,
+        upper: str,
+        lower: str = "",
+        pci_slot: str = "",
+    ) -> None:
+        self._has_secondary = lower != "" or pci_slot == ""
+        self._upper = upper
+        self._lower = lower
+        self._pci_slot = pci_slot
+        self._ip_addr = ""
+        self._mac_addr = ""
+
+    def has_secondary(self) -> bool:
+        return self._has_secondary
+
+    def __str__(self) -> str:
+        return (
+            "NicInfo:\n"
+            f"upper: {self._upper}\n"
+            f"lower: {self._lower}\n"
+            f"pci_slot: {self._pci_slot}\n"
+            f"ip_addr: {self._ip_addr}\n"
+            f"mac_addr: {self._mac_addr}\n"
+        )
+
+    def testpmd_include(self, vdev_type: str) -> str:
```

Comment:
It looks this method generate a command string for testpmd. Consider to move this method to testpmd.

1. the `descrbed_as` should be before of `is_true`, or it cannot be printed, because `is_true` failed first.
2. make the method name like "verb_noun", and straightforward. It's hard to know what happens, if called this method.
3. add comments, if it's really not easy to get the method name clear.

Reply:
Will move to testpmd, this makes it much cleaner! Good eye thank you